cmdstanpy/stanfit.py

```diff
@@ -1260,20 +1265,24 @@ class CmdStanGQ:
     Container for outputs from CmdStan generate_quantities run.
     """
 
-    def __init__(self, runset: RunSet, mcmc_sample: pd.DataFrame) -> None:
+    def __init__(
+        self,
+        runset: RunSet,
+        mcmc_sample: CmdStanMCMC,
+        logger: Optional[logging.Logger] = None,
+    ) -> None:
         """Initialize object."""
         if not runset.method == Method.GENERATE_QUANTITIES:
             raise ValueError(
                 'Wrong runset method, expecting generate_quantities runset, '
                 'found method {}'.format(runset.method)
             )
         self.runset = runset
-        self._metadata = None
+        self._logger = logger or get_logger()
```

Comment:
Some other places in this PR are moving _away_ from storing the logger, but this is still adding it. Intentional?

Reply:
thanks, oversight - will remove.